plugins/Files/js/components/filelist.js

```diff
@@ -0,0 +1,68 @@
+import React, { PropTypes } from 'react'
+import { List } from 'immutable'
+import File from './file.js'
+import Directory from './directory.js'
+import Path from 'path'
+import SearchField from '../containers/searchfield.js'
+
+const FileList = ({files, searchResults, path, showSearchField, actions}) => {
+
+	const onDirectoryClick = (directory) => () => actions.setPath(path + directory.name + '/')
```

Comment:
part of me thinks you should combine this and `onBackClick` in a new action, `actions.cdPath(dir)`, where the path manipulation logic would live.